quickwit-storage/src/lib.rs

```diff
@@ -65,12 +69,87 @@ pub use self::storage_resolver::{
     quickwit_storage_uri_resolver, StorageFactory, StorageUriResolver,
 };
 #[cfg(feature = "testsuite")]
-pub use self::tests::storage_test_suite;
+pub use self::test_suite::storage_test_suite;
 pub use crate::cache::{wrap_storage_with_long_term_cache, Cache, MemorySizedCache, SliceCache};
 pub use crate::error::{StorageError, StorageErrorKind, StorageResolverError, StorageResult};
 
+/// Returns the canonical, absolute form of a path as a URI with all intermediate
+/// components normalized and symbolic links resolved. Well-formed URIs with a protocol are returned
+/// as is. This function makes use of [`tokio::fs::canonicalize`], which accesses the local file
+/// system, and returns an error when `path_or_uri` does not exist or contains a non-final component
+/// that is not a directory.
+async fn path_to_canonical_uri(path_or_uri: &str) -> anyhow::Result<PathBuf> {
```

Comment:
I think `PathBuf` is just `alala/werewr/werwer` stuff.

It offers some nice utilities to get parents etc. but it is not meant to be a URI.

I'd return String here.